compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -3136,12 +3136,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     if self
                         .tcx
                         .all_impls(candidate.def_id)
-                        .filter(|imp_did| {
-                            self.tcx.impl_polarity(*imp_did) == ty::ImplPolarity::Negative
+                        .filter_map(|imp_did| self.tcx.impl_trait_header(imp_did))
```

Comment:
Can you turn this back into an `unwrap`? This used to implicitly serve as an assertion that we're only looking at *trait* impls? Actually, making it an `expect` is better too.

Reply:
The change in order requires removing the unwrap. Previously it was unreachable, because the negative polarity implied that it's a trait impl

Reply:
I did not test whether this is always a trait impl anyway. Should do that. I just did it out of caution